src/contracts/service/Fraction.sol

```diff
@@ -4,22 +4,65 @@ pragma solidity ^0.7.0;
 
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
 import "./BondingCurve.sol";
+//import "./CollabLandDAO.sol";
+import "moloch/contracts/Moloch.sol";
 
 contract Fraction is BondingCurve {
     bool public locked = false;
+    Moloch public dao;
+    address public summoner;
+    uint256 public periodDuration;
+    uint256 public votingPeriodLength;
+    uint256 public gracePeriodLength;
+    uint256 public proposalDeposit;
+    uint256 public dilutionBond;
+    uint256 public processingReward;    
 
     event Fungified(address nft, string name, string symbol);
 
-    constructor(string memory _name, string memory _symbol)
-        BondingCurve(_name, _symbol)
-    {}
+    constructor(
+        address _summoner, 
+        uint256 _periodDuration,
+        uint256 _votingPeriodLength,
+        uint256 _gracePeriodLength,
+        uint256 _proposalDeposit,
+        uint256 _dilutionBond,
+        uint256 _processingReward,
+        string memory _name, 
+        string memory _symbol
+    ) BondingCurve(
+        _name, 
+        _symbol
+      ) {
+        summoner = _summoner;
+        periodDuration = _periodDuration;
+        votingPeriodLength = _votingPeriodLength;
+        gracePeriodLength = _gracePeriodLength;
+        proposalDeposit = _proposalDeposit;
+        dilutionBond = _dilutionBond;
+        processingReward = _processingReward;
+    }
 
     /**
      * @param _nftids - An array of NFT ids
      */
     function fungify(ERC721 _nft, uint256[] memory _nftids) public virtual {
         require(locked == false);
 
+        address[] memory approvedTokens;
```

Comment:
When do we set `locked` to true?

Reply:
it should be called after fungify has been called, making fungify only executed once